examples/core_geometry_quaternion.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python
+
+##Copyright 2016 Jelle Feringa (jelleferinga@gmail.com)
+##
+##This file is part of pythonOCC.
+##
+##pythonOCC is free software: you can redistribute it and/or modify
+##it under the terms of the GNU Lesser General Public License as published by
+##the Free Software Foundation, either version 3 of the License, or
+##(at your option) any later version.
+##
+##pythonOCC is distributed in the hope that it will be useful,
+##but WITHOUT ANY WARRANTY; without even the implied warranty of
+##MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+##GNU Lesser General Public License for more details.
+##
+##You should have received a copy of the GNU Lesser General Public License
+##along with pythonOCC.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import print_function
+
+from functools import wraps
+
+from OCC.BRepBuilderAPI import BRepBuilderAPI_MakeEdge
+from OCC.BRepPrimAPI import BRepPrimAPI_MakeBox
+from OCC.Display.SimpleGui import init_display
+from OCC.gp import gp_QuaternionSLerp, gp_Quaternion, gp_Vec, gp_Pnt
+
+display, start_display, add_menu, add_function_to_menu = init_display()
+
+
+class assert_isdone(object):
+    '''
+    raises an assertion error when IsDone() returns false, with the error
+    specified in error_statement
+    '''
+
+    def __init__(self, to_check, error_statement):
+        self.to_check = to_check
+        self.error_statement = error_statement
+
+    def __enter__(self, ):
+        if self.to_check.IsDone():
+            pass
+        else:
+            raise AssertionError(self.error_statement)
+
+    def __exit__(self, type, value, traceback):
+        pass
+
+
+@wraps(BRepPrimAPI_MakeBox)
+def make_box(*args):
+    box = BRepPrimAPI_MakeBox(*args)
+    box.Build()
+    with assert_isdone(box, 'failed to built a cube...'):
+        return box.Shape()
+
+
+@wraps(BRepBuilderAPI_MakeEdge)
+def make_edge(*args):
+    edge = BRepBuilderAPI_MakeEdge(*args)
+    with assert_isdone(edge, 'failed to produce edge'):
+        result = edge.Edge()
+        edge.Delete()
+        return result
+
```

Comment:
should use `make_edge` from `core_geometry_utils`, perhaps `make_box` should go ther too?